lightning/src/ln/onion_utils.rs

```diff
@@ -161,11 +161,19 @@ fn xor_bufs(dst: &mut[u8], src: &[u8]) {
 	}
 }
 
+
 const ZERO:[u8; 21*65] = [0; 21*65];
-pub(super) fn construct_onion_packet(mut payloads: Vec<msgs::OnionHopData>, onion_keys: Vec<OnionKeys>, associated_data: &PaymentHash) -> msgs::OnionPacket {
+pub(super) fn construct_onion_packet(payloads: Vec<msgs::OnionHopData>, onion_keys: Vec<OnionKeys>, prng_seed: [u8; 32], associated_data: &PaymentHash) -> msgs::OnionPacket {
 	let mut buf = Vec::with_capacity(21*65);
 	buf.resize(21*65, 0);
 
+	let mut chacha = ChaCha20::new(&prng_seed, &[0; 8]);
+	chacha.process(&[0; 21*65], &mut buf);
```

Comment:
You init the filler buffer with randomness, isn't the fix about the packet, `packet_data` in `construct_onion_packet_with_init_noise` ?

Reply:
Oops, good catch! I added a quick and dirty test which would have caught this, as well.